matchbox/provider_test.go

```diff
@@ -0,0 +1,251 @@
+package matchbox
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/coreos/matchbox/matchbox/rpc"
+	"github.com/coreos/matchbox/matchbox/server"
+	"github.com/coreos/matchbox/matchbox/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var providers = map[string]terraform.ResourceProvider{
+	"matchbox": Provider(),
+}
+
+func TestProvider(t *testing.T) {
+	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+type MatchboxFixtureServer struct {
+	Store    storage.Store
+	Listener net.Listener
+}
+
+func NewMatchboxFixtureServer(s storage.Store) *MatchboxFixtureServer {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
Let's design this FixtureServer to be more clearly similar to the real server https://github.com/coreos/matchbox/blob/master/cmd/matchbox/main.go (which you're striving to emulate), as if you were writing a standalone version of the gRPC server. Same defaults, names, error message.

```
func NewFixtureServer(tlscfg TLSInfo, store storage.Store) *MatchboxFixtureServer {
    return &MatchboxFixtureServer{
        TLSInfo: tlsinfo,
        Storage: store,
    }
}
```